packages/record-data/addon/-private/relationships/state/relationship.ts

```diff
@@ -1,14 +1,33 @@
-import { guidFor } from '@ember/object/internals';
 import { get } from '@ember/object';
-import { relationshipStateFor, implicitRelationshipStateFor } from '../../record-data-for';
+import { relationshipStateFor, implicitRelationshipStateFor, implicitRelationshipsFor } from '../../record-data-for';
 import { assert, warn } from '@ember/debug';
 import OrderedSet from '../../ordered-set';
 import _normalizeLink from '../../normalize-link';
-import { RelationshipRecordData } from '../../ts-interfaces/relationship-record-data';
 import { JsonApiRelationship } from '@ember-data/store/-private/ts-interfaces/record-data-json-api';
 import { RelationshipSchema } from '@ember-data/store/-private/ts-interfaces/record-data-schemas';
 import { CUSTOM_MODEL_CLASS } from '@ember-data/canary-features';
 import { PaginationLinks } from '@ember-data/store/-private/ts-interfaces/ember-data-json-api';
+import { StableRecordIdentifier } from '@ember-data/store/-private/ts-interfaces/identifier';
+import { RelationshipRecordData } from '../../ts-interfaces/relationship-record-data';
+import { RecordData } from '@ember-data/store/-private/ts-interfaces/record-data';
+import CoreStore from '@ember-data/store/-private/system/core-store';
+import { recordDataFor as peekRecordData } from '@ember-data/store/-private';
+
+type RecordDataStoreWrapper = import('@ember-data/store/-private/system/store/record-data-store-wrapper').default;
+
+export function isNew(identifier: StableRecordIdentifier): boolean {
+  if (!identifier.id) {
+    return true;
+  }
+  const recordData = peekRecordData(identifier);
```

Comment:
Can we leave a comment here as to the logic/sideeffect this uses, a couple tests to confirm the behavior might also be great but wouldn't want to block but at least file an issue please